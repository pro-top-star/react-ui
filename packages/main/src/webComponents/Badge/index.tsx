import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Badge from '@ui5/webcomponents/dist/Badge';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface BadgePropTypes extends WithWebComponentPropTypes {
  colorScheme?: string; // @generated
  children?: string; // @generated
  icon?: any; // @generated
}

/**
 * <code>import { Badge } from '@ui5/webcomponents-react/lib/Badge';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Badge" target="_blank">UI5 Web Components Playground</a>
 */
const Badge: FC<BadgePropTypes> = withWebComponent<BadgePropTypes>(UI5Badge);

Badge.displayName = 'Badge';

Badge.defaultProps = {
  colorScheme: '1' // @generated
};

export { Badge };
