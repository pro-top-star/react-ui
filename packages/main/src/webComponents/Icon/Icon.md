[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Icon)

```jsx
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
```
