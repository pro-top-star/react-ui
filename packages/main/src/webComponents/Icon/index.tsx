import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Icon from '@ui5/webcomponents/dist/Icon';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface IconPropTypes extends WithWebComponentPropTypes {
  accessibleName?: string; //@generated
  name?: string; // @generated
  showTooltip?: boolean; // @generated
}

const Icon: FC<IconPropTypes> = withWebComponent<IconPropTypes>(UI5Icon);

Icon.defaultProps = {
  showTooltip: false
};

Icon.displayName = 'Icon';

export { Icon };
