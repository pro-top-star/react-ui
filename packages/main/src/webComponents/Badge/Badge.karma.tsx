import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { Badge } from '../../lib/Badge';
import { Icon } from '../../lib/Icon';

use(matchSnapshot);

describe('Badge', () => {
  it('Basic Test', () => {
    const wrapper = mountThemedComponent(<Badge>My Badge</Badge>);
    expect(wrapper.debug()).to.matchSnapshot();
  });

  it('with Icon', () => {
    const wrapper = mountThemedComponent(<Badge icon={<Icon src="sap-icon://employee" />}>My Badge</Badge>);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
