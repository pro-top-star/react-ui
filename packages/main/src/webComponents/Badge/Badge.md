[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Badge)

```jsx
import { Badge } from '@ui5/webcomponents-react/lib/Badge';
```
