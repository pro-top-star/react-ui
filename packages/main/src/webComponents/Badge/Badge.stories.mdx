import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Badge } from '@ui5/webcomponents-react/lib/Badge';
import '@ui5/webcomponents-icons/dist/icons/employee.js';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import { CSSProperties, Ref } from 'react';

import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';

<Meta
  title="UI5 Web Components / Badge"
  component={Badge}
  argTypes={{
    ...createSelectArgTypes({}),
    children: { control: 'text' },
    icon: { control: { disable: true } },
    slot: { control: { disable: true } },
    ref: { control: { disable: true } },
    style: {
      type: CSSProperties,
      description:
        'Element style which will be appended to the most outer element of a component. Use this prop carefully, some css properties might break the component.'
    },
    className: {
      type: 'string',
      description:
        'CSS Class Name which will be appended to the most outer element of a component. Use this prop carefully, overwriting CSS rules might break the component.'
    },
    tooltip: { type: 'string', description: 'A tooltip which will be shown on hover' }
  }}
  args={{
    children: 'Badge Text',
    icon: <Icon name="employee" />,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Badge {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
<div style={{fontFamily:'var(--sapFontFamily)', fontSize:'var(--sapFontSize)', color:'var(--sapTextColor)'}}>
<h3>Usage Guidelines</h3>
<ul>
  <li>If the text is longer than the width of the component, it doesn’t wrap, it shows ellipsis.</li>
  <li>
    When truncated, the full text is not visible, therefore, it’s recommended to make more space for longer texts to be
    fully displayed.
  </li>
  <li>Colors are not semantic and have no visual representation in High Contrast Black (sap_belize_hcb) theme.</li>
</ul>
</div>