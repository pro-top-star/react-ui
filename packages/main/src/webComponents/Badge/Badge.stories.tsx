import { text } from '@storybook/addon-knobs';
import { Badge } from '@ui5/webcomponents-react/lib/Badge';
import React from 'react';

export default {
  title: 'UI5 Web Components / Badge',
  component: Badge
};

export const generatedDefaultStory = () => (
  <Badge colorScheme={text('colorScheme', '1')} icon={null}>
    Some Content
  </Badge>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
