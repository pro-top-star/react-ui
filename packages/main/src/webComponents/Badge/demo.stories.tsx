import { select } from '@storybook/addon-knobs';
import '@ui5/webcomponents-icons/dist/icons/employee';
import { Badge } from '@ui5/webcomponents-react/lib/Badge';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import React from 'react';
import notes from './Badge.md';

export default {
  title: 'UI5 Web Components / Badge',
  component: Badge,
  parameters: {
    notes
  }
};

export const defaultStory = () => (
  <Badge
    colorScheme={select('colorScheme', ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'], '4')}
    icon={<Icon name="employee" />}
  >
    Badge
  </Badge>
);

defaultStory.story = {
  name: 'Default story'
};
