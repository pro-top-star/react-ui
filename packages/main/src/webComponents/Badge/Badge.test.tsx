import { mountThemedComponent } from '@shared/tests/utils';
import { Badge } from '@ui5/webcomponents-react/lib/Badge';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import React from 'react';

describe('Badge', () => {
  test('Basic Test', () => {
    const wrapper = mountThemedComponent(<Badge>My Badge</Badge>);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('with Icon', () => {
    const wrapper = mountThemedComponent(<Badge icon={<Icon name="employee" />}>My Badge</Badge>);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
