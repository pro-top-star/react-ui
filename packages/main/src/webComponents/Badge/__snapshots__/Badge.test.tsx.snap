// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Basic Test 1`] = `
<ui5-badge
  color-scheme="1"
>
  My Badge
</ui5-badge>
`;

exports[`Badge with Icon 1`] = `
<ui5-badge
  color-scheme="1"
>
  <ui5-icon
    name="employee"
    slot="icon"
  />
  My Badge
</ui5-badge>
`;
